File: backend/src/main/java/com/flowlink/core/repository/MockServerRepository.java
package com.flowlink.core.repository;

import com.flowlink.core.domain.MockServer;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface MockServerRepository extends JpaRepository<MockServer, UUID> {

    List<MockServer> findByTenantIdOrderByUpdatedAtDesc(String tenantId);

    Optional<MockServer> findByIdAndTenantId(UUID id, String tenantId);

    /** 서빙 경로 조회 — slug 는 전역 유니크라 테넌트 무관(무인증 게이트웨이용). */
    Optional<MockServer> findBySlug(String slug);

    boolean existsBySlug(String slug);
}

File: backend/src/main/kotlin/com/flowlink/core/repository/MockServerRepository.kt
package com.flowlink.core.repository

import com.flowlink.core.domain.MockServer
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.Optional
import java.util.UUID

interface MockServerRepository : JpaRepository<MockServer, UUID> {

    /** 워크스페이스 삭제 시 안의 mock 이관 — 유령(관리 불가·서빙 지속) 방지. to=삭제 실행자 개인 ws, null=공용 폴백. */
    @Modifying
    @Query("update MockServer m set m.workspaceId = :to where m.workspaceId = :from")
    fun reassignWorkspace(@Param("from") from: UUID, @Param("to") to: UUID?): Int

    fun findByTenantIdOrderByUpdatedAtDesc(tenantId: String): List<MockServer>

    fun findByIdAndTenantId(id: UUID, tenantId: String): Optional<MockServer>

    /** 서빙 경로 조회 — slug 는 팀 스코프 유니크(무인증 게이트웨이는 경로의 tenant 세그먼트로 조회). */
    fun findByTenantIdAndSlug(tenantId: String, slug: String): Optional<MockServer>

    fun existsByTenantIdAndSlug(tenantId: String, slug: String): Boolean
}
